qualification/recv.py

```diff
@@ -92,6 +92,10 @@ def __init__(self, cbf: CBFRemoteControl, stream_names: Sequence[str]) -> None:
         self.cbf = cbf
         self._acv_name = acv_name
 
+    def is_complete_chunk(self, chunk: katgpucbf.recv.Chunk) -> bool:
+        """Check whether this chunk is complete (no missing data)."""
+        return bool(np.all(chunk.present))
```

Comment:
I gather casting it to a plain Python bool has done wonders for speeding up the check. Interesting indeed.

Reply:
No, that cast is just to keep mypy happy. The big speedup is in the BaselineCorrelationProductsReceiver.is_complete_chunk overload where I now only check the first channel in each heap.

Reply:
Aha, I realised this after reading your updates on the ticket. Really, excellent.